### PR TITLE
cxxtools: 2.2.1 -> unstable-2021-12-23; tntnet: 2.2.1 -> unstable-2021-03-29

### PR DESCRIPTION
**Pull request by armeenm**
###### Description of changes
ZHF: #172160 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by armeenm** [comment]
Seems tntnet needs a bump too...

**Comment by risicle** [comment]
Think the `cxxtools` homepage also needs updating.

**Comment by armeenm** [comment]
Should be good to go now.

**Comment by juliendehos** [comment]
Thank you for updating these packages (I don't use them anymore and I forgot to update them :sweat:).
Maybe it would be better to use V3.0 instead of unstable ?

**Comment by armeenm** [comment]
I tried but tntnet v3 failed to build with some syntax errors. Trunk tntnet doesn't build with cxxtools v3, so that had to get bumped as well.

**Comment by juliendehos** [comment]
Oh, ok.
Tntnet does not seem very active now. So another solution would be updating cxxtools to V3.0 and removing tntnet and tntdb. I don't know which solution is the best one.

**Comment by risicle** [comment]
Hmm fails to build for me on linux aarch64:

```
builder for '/nix/store/kxqfnbvsy6vpxmrbzcywfnz32a9nfq4s-cxxtools-unstable-2021-12-23.drv' failed with exit code 2; last 10 log lines:
  g++ -DHAVE_CONFIG_H -I. -I../src  -I../src -I../include -I../include -Wno-long-long -Wall -pedantic  -g -O2  -c -o cxxtz.o cxxtz.cpp
  /nix/store/qrw4ldkpsgbk1gzwf45qz40l43ld79z4-bash-5.1-p16/bin/bash ../libtool  --tag=CXX   --mode=link g++  -g -O2    -o cxxtz cxxtz.o ../src/libcxxtools.la -lrt  -lpthread
  libtool: link: g++ -g -O2 -o .libs/cxxtz cxxtz.o  ../src/.libs/libcxxtools.so -lssl /nix/store/w8igxmgl8snapf6kraxvc666r6c30j8x-gcc-10.3.0-lib/lib/../lib64/libstdc++.so -lm -lrt -lpthread -Wl,-rpath -Wl,/nix/store/483y4l7853x9zaybicjqwnwinlbkw3c1-cxxtools-unstable-2021-12-23/lib -Wl,-rpath -Wl,/nix/store/w8igxmgl8snapf6kraxvc666r6c30j8x-gcc-10.3.0-lib/lib/../lib64
  /nix/store/kgjkrgjh4xclqlilirknklkvqmp9yx7c-binutils-2.38/bin/ld: .libs/cxxtz: hidden symbol `__aarch64_ldadd4_acq_rel' in /nix/store/5hp6scld4p1zx1kism1vssvyrbm9cyvm-gcc-10.3.0/lib/gcc/aarch64-unknown-linux-gnu/10.3.0/libgcc.a(ldadd_4_4.o) is referenced by DSO
  /nix/store/kgjkrgjh4xclqlilirknklkvqmp9yx7c-binutils-2.38/bin/ld: final link failed: bad value
  collect2: error: ld returned 1 exit status
  make[1]: *** [Makefile:420: cxxtz] Error 1
  make[1]: *** Waiting for unfinished jobs....
  make[1]: Leaving directory '/build/source/tools'
  make: *** [Makefile:542: all-recursive] Error 1
```

**Comment by armeenm** [comment]
Hm. Shame I don't have access to anything arm64 at the moment to investigate.

**Comment by risicle** [comment]
https://nixos.wiki/wiki/NixOS_on_ARM#Compiling_through_binfmt_QEMU

**Comment by Artturin** [comment]
the failure might have something to do with the differing gcc versions

```
nix-repl> legacyPackages.aarch64-linux.gcc
«derivation /nix/store/f3bmmfa0yjdwbcg26d5g6j0pcw6rb0dw-gcc-wrapper-9.3.0.drv»

nix-repl> legacyPackages.x86_64-linux.gcc
«derivation /nix/store/3f5j2vnm9rw8j4vdnmgp69za10k697zc-gcc-wrapper-11.3.0.drv»
```

**Review by Artturin — DISMISSED** [review]
no build regressions

**Comment by Artturin** [comment]
```
./results/cxxtools/bin/cxxtz --version
failed to open timezone file "/etc/zoneinfo//nix/store/by7piy54rpa2ql165sb7ifcajs8axa7x-tzdata-2022a/share/zoneinfo/Europe/Helsinki"
```

```
/results/cxxtools/bin/cxxtz
Usage: ./results/cxxtools/bin/cxxtz {options} {times}

Synopsis:
current time in time zone Pacific/Nauru
 ./results/cxxtools/bin/cxxtz -c -z Pacific/Nauru
dump information about the current time zone
 ./results/cxxtools/bin/cxxtz -d
output some GMT times in the time zone
 ./results/cxxtools/bin/cxxtz '2017-03-26 01:00:00' '2017-03-26 00:59:59'

Options:
 -a             list all valid time zones
 -z <timezone>  specify time zone (default: system time zone)
 -Z             output time zone of the system
 -d             dump time zone informations
 -c             output current time in the specified time zone
 -u             print current utc time
 -r             reverse
 -e             early flag (with -r take earlier time if ambiguous)
 -l             late flag (with -r take later time if ambiguous)
 -v             output previous and next change of dst
failed to open timezone file "/etc/zoneinfo//nix/store/by7piy54rpa2ql165sb7ifcajs8axa7x-tzdata-2022a/share/zoneinfo/Europe/Helsinki"

```

File: pkgs/development/libraries/cxxtools/default.nix [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
+    owner = "maekitalo";
+    repo = "cxxtools";
+    rev = "31a212fe400b36cc5b9ea0dd76d9b5facfde914d";
+    hash = "sha256-h7/G5aPnEsGjOL2NrKAnw4VfWTr8o4u574albdqFdjg=";
   };
 
+  nativeBuildInputs = [ autoreconfHook ];
+  buildInputs = [ openssl.dev ];
+
   configureFlags = lib.optional stdenv.isAarch64 "--with-atomictype=pthread";
```
Comment (sikmir) [inline pkgs/development/libraries/cxxtools/default.nix:21]: I guess we don't need it anymore, see https://github.com/maekitalo/cxxtools/commit/c5baa0b7756ee88f51ec58fe9a4d3b416266cc9f

File: pkgs/development/libraries/cxxtools/default.nix [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
-  meta = {
-    homepage = "http://www.tntnet.org/cxxtools.html";
+  meta = with lib; {
+    homepage = "https://github.com/maekitalo/cxxtools";
     description = "Comprehensive C++ class library for Unix and Linux";
-    platforms = lib.platforms.linux ;
-    license = lib.licenses.lgpl21;
-    maintainers = [ lib.maintainers.juliendehos ];
+    platforms = platforms.linux;
+    license = licenses.lgpl21;
```
Comment (sikmir) [inline pkgs/development/libraries/cxxtools/default.nix:29]:
```suggestion
    license = licenses.lgpl21Plus;
```

**Comment by sikmir** [comment]
> I tried but tntnet v3 failed to build with some syntax errors. Trunk tntnet doesn't build with cxxtools v3, so that had to get bumped as well.

tntnet v3 with this patch https://github.com/maekitalo/tntnet/commit/69adfc8ee351a0e82990c1ffa7af6dab726e1e49 (autoconf-2.70 fixes) is fine.

**Pull request closed without merging**